File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest43251.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest43251 {

    @GetMapping("/BenchmarkTest43251")
    public void BenchmarkTest43251(@RequestParam("id") String id, HttpServletRequest request, HttpServletResponse response) throws Exception {
        response.sendError(403, "directory listing forbidden");
    }
}

File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_43251.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


async def BenchmarkTest43251(request: Request):
    path_value = request.path_params.get('id', '')
    data = str(path_value).replace('\x00', '')
    if not str(data).isdigit():
        raise Exception('error: ' + str(data))
    return {"updated": True}
